### ISSUE TITLE
Should the delay server shutdown if its parent disappears?

### ISSUE THREAD

**Issue by korydraughn** [irods/irods#6345]
Things to consider:
- The delay server will run forever until it is signaled to terminate (e.g. `SIGTERM`)
- The parent iRODS process will automatically relaunch the delay server if it determines it is supposed to be running
- The parent iRODS process requires that the delay server be a child process because of the delay server migration
- The delay server is a client application (i.e. fork-exec model)
    - Are there benefits to the delay server not being managed by the iRODS server?

**Comment by kript** [comment]
Should/could this be handled by systemd? Is there a use case for having different systemd script for rodsServer and reServer?

**Comment by korydraughn** [comment]
The issue with using systemd is that the delay server migration code requires that the delay server be a child of the original iRODS process.

Q. What would an admin expect to happen if the parent iRODS process goes away and only the delay server is left running?

**Comment by trel** [comment]
Tying our processes together makes it easier to manage/understand - but comes with the downside that they have to be very smart / restartable / refreshable, independent of the OS tools available (systemd).

> Q. What would an admin expect to happen if the parent iRODS process goes away and only the delay server is left running?

Maybe a standalone delay server would be smart enough to detect whether it has a companion iRODS server?   But how is that really any different that the coordination we do now with parent/child processes?   And would it require a separate configuration file?

---

Yes, if a delay server sees its parent go away - I think it should exit as gracefully as possible.   With the current model, as soon as a new iRODS server stands up, it will stand up its own delay server.   And there should be only one.

**Comment by korydraughn** [comment]
I agree that fully separating the delay server from the iRODS server does not present itself as being very beneficial. Doing that would likely increase complexity.

As for the delay server tracking its parent, I agree that it is probably the right choice to have it shutdown. Under normal operation, this makes the most sense. We can't guard against the OS doing something weird resulting in the iRODS server disappearing. At this point, the admin must kill the delay server before starting the iRODS server again.

**Comment by trel** [comment]
okay, so we'll leave this issue for that work.   marking as enhancement.

**Comment by trel** [comment]
@korydraughn ... is this handled by the new process model in 5.0 (upcoming)?

**Comment by korydraughn** [comment]
Not yet, but easy enough to add.

Will consider whether the approach outlined here is the right thing to do.

**Comment by trel** [comment]
moving to 5.2.0 as it seems useful/good.
